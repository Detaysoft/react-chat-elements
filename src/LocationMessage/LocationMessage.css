.rce-mbox-location {
    position: relative;
    width: 250px;
    overflow: hidden;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-top: -3px;
    margin-right: -6px;
    margin-left: -6px;
    border-radius: 5px;
}

.rce-mbox-location-img {
    width: 100%;
}

.rce-mbox-location-text {
  width: 250px;
  margin-left: -6px;
  margin-right: -6px;
}
