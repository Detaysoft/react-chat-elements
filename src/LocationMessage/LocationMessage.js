import React, { Component } from 'react';
import './LocationMessage.css';

const classNames = require('classnames');

const STATIC_URL = 'https://maps.googleapis.com/maps/api/staticmap?markers=color:MARKER_COLOR|LATITUDE,LONGITUDE&zoom=ZOOM&size=270x200&scale=2&key=KEY';
const MAP_URL = 'https://www.google.com/maps/search/?api=1&query=LATITUDE,LONGITUDE&zoom=ZOOM';

export class LocationMessage extends Component {
    constructor(props) {
        super(props);

        this.className = this.className.bind(this);
    }

    buildURL(url) {
        var center = this.props.data || {};

        return url.replace('LATITUDE', center.latitude)
                  .replace('LONGITUDE', center.longitude)
                  .replace('MARKER_COLOR', this.props.markerColor)
                  .replace('ZOOM', this.props.zoom)
                  .replace('KEY', this.props.apiKey);
    }

    className() {
        var className = classNames('rce-mbox-location', this.props.className);

        if (this.props.text) {
            className = classNames(className, 'rce-mbox-location-has-text');
        }

        return className;
    }

    render() {
        return (
            <div className='rce-container-lmsg'>
                <a
                    onClick={this.props.onOpen}
                    target={this.props.target}
                    href={this.props.href ||Â this.props.src || this.buildURL(MAP_URL)}
                    className={this.className()}>
                    <img
                        onError={this.props.onError}
                        className='rce-mbox-location-img'
                        src={
                            this.props.src ||
                            this.buildURL(STATIC_URL)
                        }/>
                </a>
                {
                    this.props.text &&
                    <div className="rce-mbox-text rce-mbox-location-text">
                        {this.props.text}
                    </div>
                }
            </div>
        );
    }
}

LocationMessage.defaultProps = {
    target: '_blank',
    apiKey: '',
    zoom: 14,
    markerColor: 'red',
    onError: () => void(0),
}

export default LocationMessage;
