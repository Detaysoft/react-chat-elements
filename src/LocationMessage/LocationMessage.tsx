import './LocationMessage.css'
import classNames from 'classnames'
import { ILocationMessageProps } from '../type'

const STATIC_URL =
  'https://maps.googleapis.com/maps/api/staticmap?markers=color:MARKER_COLOR|LATITUDE,LONGITUDE&zoom=ZOOM&size=270x200&scale=2&key=KEY'
const MAP_URL = 'https://www.google.com/maps/search/?api=1&query=LATITUDE,LONGITUDE&zoom=ZOOM'

const LocationMessage: React.FC<ILocationMessageProps> = ({ markerColor = 'red', target = '_blank', zoom = '14', ...props }) => {
  const buildURL = (url: string) => {
    return url
      .replace(/LATITUDE/g, props?.data.latitude)
      .replace(/LONGITUDE/g, props?.data.longitude)
      .replace('MARKER_COLOR', markerColor)
      .replace('ZOOM', zoom)
      .replace('KEY', props.apiKey)
  }
  const className = () => {
    let _className = classNames('rce-mbox-location' as any, props.className as any)

    if (props.text) {
      _className = classNames(_className as any, 'rce-mbox-location-has-text' as any)
    }

    return _className
  }

  return (
    <div className='rce-container-lmsg'>
      <a
        onClick={props.onOpen}
        target={target}
        href={props.href || props.src || buildURL(props.data.mapURL || MAP_URL)}
        className={className()}
      >
        <img
          onError={props.onError}
          className='rce-mbox-location-img'
          src={props.src || buildURL(props.data.staticURL || STATIC_URL)}
        />
      </a>
      {props.text && <div className='rce-mbox-text rce-mbox-location-text'>{props.text}</div>}
    </div>
  )
}

export default LocationMessage
