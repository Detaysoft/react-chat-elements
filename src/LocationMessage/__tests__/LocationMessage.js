import React, { Component } from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LocationMessage from '../LocationMessage'

describe('LocationMessage component', () => {
  it('should render without issues', () => {
    const component = shallow(<LocationMessage data={{ latitude: '', longitude: '', staticURL: '' }} />)
    expect(component.length).toBe(1)
    expect(toJson(component)).toMatchSnapshot()
  })
})
