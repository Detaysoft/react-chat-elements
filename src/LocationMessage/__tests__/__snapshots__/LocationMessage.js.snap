// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationMessage component should render without issues 1`] = `
<div
  className="rce-container-lmsg"
>
  <a
    className="rce-mbox-location"
    href="https://www.google.com/maps/search/?api=1&query=undefined,undefined&zoom=14"
    target="_blank"
  >
    <img
      className="rce-mbox-location-img"
      onError={[Function]}
      src="https://maps.googleapis.com/maps/api/staticmap?markers=color:red|undefined,undefined&zoom=14&size=270x200&scale=2&key="
    />
  </a>
</div>
`;
